client/src/main/scala/org/apache/celeborn/client/ChangePartitionManager.scala

```diff
@@ -79,14 +82,18 @@ class ChangePartitionManager(
                 batchHandleChangePartitionExecutors.submit {
                   new Runnable {
                     override def run(): Unit = {
-                      val distinctPartitions = requests.synchronized {
-                        // For each partition only need handle one request
-                        requests.asScala.filter { case (partitionId, _) =>
-                          !inBatchPartitions.get(shuffleId).contains(partitionId)
-                        }.map { case (partitionId, request) =>
-                          inBatchPartitions.get(shuffleId).add(partitionId)
-                          request.asScala.toArray.maxBy(_.epoch)
-                        }.toArray
+                      val distinctPartitions = {
+                        val requestSet = inBatchPartitions.get(shuffleId)
+                        requests.asScala.map { case (partitionId, request) =>
+                          locks(partitionId % locks.length).synchronized {
+                            if (!requestSet.contains(partitionId)) {
```

Comment:
`contains` for `ConcurrentHashMap` is actually `containsValue`. It's better to use `ConcurrentHashMap.newKeySet()` instead of `ConcurrentHashMap[Integer, Unit]`

Reply:
oops, fixed